ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/openlayers/map.openlayers.js

```diff
@@ -632,24 +649,40 @@ module.exports = function OpenlayersMap(
       return this.createVectorLayer(locationModel, feature)
     },
     showMultiPolygonShape: function(locationModel) {
-      let lineObject = locationModel
+      const multiPolyObject = locationModel
         .get('multipolygon')
         .map(poly =>
           poly.map(coordinateSet =>
-            coordinateSet.map(pair => convertPointCoordinate(pair))
+            coordinateSet.map(pair => unconvertPointCoordinate(pair))
           )
         )
 
-      let feature = new Openlayers.Feature({
-        geometry: new Openlayers.geom.MultiPolygon(lineObject),
+      const bufferWidth =
+        DistanceUtils.getDistanceInMeters(
+          locationModel.get('polygonBufferWidth'),
+          locationModel.get('polygonBufferUnits')
+        ) || DEFAULT_BUFFER
+
+      const multiPolygon = Turf.multiPolygon(multiPolyObject)
+
+      const buffered = Turf.buffer(multiPolygon, bufferWidth, 'meters')
+
+      const convertedCoordinates = buffered.geometry.coordinates.map(poly =>
+        poly.map(coordinateSet =>
+          coordinateSet.map(coord => convertPointCoordinate(coord))
+        )
+      )
+
+      const feature = new Openlayers.Feature({
+        geometry: new Openlayers.geom.MultiPolygon(convertedCoordinates),
```

Comment:
You could probably refactor the code for `showMultiPolygonShape` and `showMultiLineShape`